src/pandapipes/component_models/valve_component.py

```diff
@@ -36,6 +36,64 @@ def active_identifier(cls):
     def get_connected_node_type(cls):
         return Junction
 
+    @classmethod
+    def internal_node_name(cls):
+        return "valve_nodes"
+
+    @classmethod
+    def get_internal_node_number(cls, net, return_internal_only=True):
+        """
+
+        :param net: The pandapipes network
+        :type net: pandapipesNet
+        :return:
+        :rtype:
+        """
+
+        int_nodes = np.zeros(len(net[cls.table_name()]), dtype=np.int32)
+        mask_p = np.flatnonzero(net[cls.table_name()]['et'].values == 'p')
+        val = net[cls.table_name()][['junction', 'element']].iloc[mask_p].values
+        _, idx, inv = np.unique(val, return_index=True, return_inverse=True, axis=0)
+        int_nodes[mask_p[idx]] = 1
```

Comment:
See my comment below: Is this the lookup we need to check on the array-level for e.g. "active" and the "from_junction"?